lib/oxidized/model/asa.rb

```diff
@@ -27,27 +34,53 @@ class ASA < Oxidized::Model
     comment cfg
   end
 
-  cmd 'more system:running-config' do |cfg|
-    cfg = cfg.each_line.to_a[3..-1].join
-    cfg.gsub! /^: [^\n]*\n/, ''
-    # backup any xml referenced in the configuration.
-    anyconnect_profiles = cfg.scan(Regexp.new('(\sdisk0:/.+\.xml)')).flatten
-    anyconnect_profiles.each do |profile|
-  	  cfg << (comment profile + "\n" )
-   	  cmd ("more" + profile) do |xml|
-	      cfg << (comment xml)
-	    end
-    end
-    # if DAP is enabled, also backup dap.xml
-    if cfg.rindex(/dynamic-access-policy-record\s(?!DfltAccessPolicy)/)
-   	  cfg << (comment "disk0:/dap.xml\n")
-      cmd "more disk0:/dap.xml" do |xml|
-        cfg << (comment xml)
+  post do
+    if (@context == true)
```

Comment:
this seems like too much code inside a single method. maybe we could think about refactoring this to something like:

``` ruby
post do
  if @context
    multiple_contexts
  else
    single_context
  end
end

private

def multiple_contexts
  # Multiple context mode
  cmd 'changeto system' do |cfg|
    cmd 'show running-config' do |systemcfg|
    allcfg = "\n\n" + systemcfg + "\n\n"
    contexts = systemcfg.scan(/^context (\S+)$/)
    files = systemcfg.scan(/config-url (\S+)$/)
    # ...
end
```

what do you guys think? /cc @ytti + @panzerfly